src/vs/vscode.d.ts

```diff
@@ -6002,9 +6052,10 @@ declare module 'vscode' {
 		placeholder: string;
 
 		/**
-		 * The warning threshold for lines in the input box.
+		 * A validation provider for the input box. It's possible to change
+		 * the validation provider simply by setting this property to a different value.
 		 */
-		lineWarningLength: number | undefined;
+		validationProvider: SourceControlInputBoxValidationProvider;
```

Comment:
Yeah, maybe... It doesn't fill well into the provider approach which usually is about registering multiple providers, like `registerInputBoxValidationProvider('git', myProviderInstance)`. I'd say make it a real provider or let be a simple callback.